components/api_request_helper/api_request_helper_unittest.cc

```diff
@@ -145,16 +131,15 @@ TEST_F(ApiRequestHelperUnitTest, SanitizedRequest) {
       "{\"id\":1,\"jsonrpc\":\"2.0\",\"result\":1.8446744073709552e+19}";
   std::string server_raw_response =
       "{\"id\":1,\"jsonrpc\":\"2.0\",\"result\":18446744073709551615}";
-  SendRequest(server_raw_response, expected_sanitized_response,
-              ParseJson(expected_sanitized_response));
-  SendRequest("", "", base::Value());
-  SendRequest("{}", "{}", base::Value(base::Value::Type::DICT));
-  SendRequest("{", "", base::Value());
-  SendRequest("0", "", base::Value());
-  SendRequest("a", "", base::Value());
+  SendRequest(server_raw_response, ParseJson(expected_sanitized_response));
+  SendRequest("", base::Value());
+  SendRequest("{}", base::Value(base::Value::Type::DICT));
+  SendRequest("{", base::Value());
+  SendRequest("0", base::Value());
+  SendRequest("a", base::Value());
   // Android's sanitizer doesn't support trailing commas.
 #if !BUILDFLAG(IS_ANDROID)
```

Comment:
It's actually weird that we allow such difference. Imagine json coming from component which works on one platform but fails on another.
But out of scope of this PR

Reply:
Yep, it was a surprise. We changed the parsing logic here, but android java parser doesn't care:
https://github.com/brave/brave-core/blob/optimize-request-api-helper-base-value-processing/chromium_src/services/data_decoder/public/cpp/data_decoder.cc#L10